src/swift/Data.swift

```diff
@@ -38,51 +43,50 @@ public struct DispatchData : RandomAccessCollection {
 			}
 		}
 	}
-#endif
-	internal var __wrapped: dispatch_data_t
+
+	internal var __wrapped: __DispatchData
 
 	/// Initialize a `Data` with copied memory content.
 	///
 	/// - parameter bytes: A pointer to the memory. It will be copied.
 	/// - parameter count: The number of bytes to copy.
 	public init(bytes buffer: UnsafeBufferPointer<UInt8>) {
-		__wrapped = dispatch_data_create(
-			buffer.baseAddress!, buffer.count, nil, _dispatch_data_destructor_default())
+		let d = dispatch_data_create(buffer.baseAddress!, buffer.count, nil, _dispatch_data_destructor_default())
+		self.init(data: d)
```

Comment:
Compiler doesn't complain.  I think it is allowed because this init is public and the init being calling with self.init is internal.  (or there's a compiler bug...)